### ISSUE TITLE
bug: Ionic core throwing error after stencil got updated yesterday

### ISSUE THREAD

**Issue by liho00** [ionic-team/ionic-framework#24153]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

It was fine before yesterday. Now unable to run the project because these error. I suspect the stencil got updated yesterday causing this error, however i cannot fix this by setting my package.json, because the error comming from @ionic/core's node_modules.

![image](https://user-images.githubusercontent.com/77238199/139925143-19c3a2c7-1c27-4956-85e1-e4c36798d970.png)

full error logs,
```
> Executing task: npm run dev <


> @0.0.0 dev /Users/liho/Desktop/349assistant
> vite

mode development
Pre-bundling dependencies:
  vue
  pinia
  @ionic/vue
  @capacitor/core
  @capacitor/splash-screen
  (...and 27 more)
(this will be run only when your dependencies or config have changed)
 > node_modules/@ionic/core/components/ion-accordion.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-accordion-group.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-avatar.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-backdrop.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-badge.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-breadcrumb.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-breadcrumbs.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-button.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-card.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-card-header.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-card-subtitle.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-card-title.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-checkbox.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-chip.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-col.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, forceUpdate, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-content.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, readTask, forceUpdate, h, Host, proxyCustomElement } from '@stencil/core/inte...
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-datetime.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, writeTask, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-fab.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-fab-button.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-fab-list.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-grid.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-img.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-item.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, forceUpdate, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-item-divider.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-item-option.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-list-header.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-menu.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, Build, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-menu-button.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-menu-toggle.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-nav.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Build, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-note.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-progress-bar.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-radio.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-range.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-reorder.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-ripple-effect.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, readTask, writeTask, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-row.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-segment.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, writeTask, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-segment-button.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, forceUpdate, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-select.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-select-option.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-skeleton-text.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-spinner.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-split-pane.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, Build, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-text.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-thumbnail.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-title.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-toggle.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-toolbar.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, forceUpdate, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-back-button.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-router-outlet.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-tab-button.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-tab-bar.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-toast.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-modal.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, writeTask, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

 > node_modules/@ionic/core/components/ion-popover.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
    1 │ import { attachShadow, createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';
      ╵          ~~~~~~~~~~~~

error when starting dev server:
Error: Build failed with 56 errors:
node_modules/@ionic/core/components/ion-accordion-group.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
node_modules/@ionic/core/components/ion-accordion.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
node_modules/@ionic/core/components/ion-avatar.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
node_modules/@ionic/core/components/ion-back-button.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
node_modules/@ionic/core/components/ion-backdrop.js:1:9: error: No matching export in "node_modules/@ionic/core/node_modules/@stencil/core/internal/client/index.js" for import "attachShadow"
...
    at failureErrorWithLog (/Users/liho/Desktop/349assistant/node_modules/esbuild/lib/main.js:1475:15)
    at /Users/liho/Desktop/349assistant/node_modules/esbuild/lib/main.js:1133:28
    at runOnEndCallbacks (/Users/liho/Desktop/349assistant/node_modules/esbuild/lib/main.js:923:63)
    at buildResponseToResult (/Users/liho/Desktop/349assistant/node_modules/esbuild/lib/main.js:1131:7)
    at /Users/liho/Desktop/349assistant/node_modules/esbuild/lib/main.js:1240:14
    at /Users/liho/Desktop/349assistant/node_modules/esbuild/lib/main.js:611:9
    at handleIncomingPacket (/Users/liho/Desktop/349assistant/node_modules/esbuild/lib/main.js:708:9)
    at Socket.readFromStdout (/Users/liho/Desktop/349assistant/node_modules/esbuild/lib/main.js:578:7)
    at Socket.emit (events.js:375:28)
    at Socket.emit (domain.js:470:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @0.0.0 dev: `vite`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @0.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/liho/.npm/_logs/2021-11-02T18_37_17_883Z-debug.log
The terminal process "zsh '-c', 'npm run dev'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.
```

### Expected Behavior

The ionic core should persist the stencil version, prevent this error happend again..

### Steps to Reproduce

```
    "@ionic/vue": "^6.0.0-beta.6",
    "@ionic/vue-router": "^6.0.0-beta.6",
    "vue": "^3.1.5",
    "vite": "2.5.10",
```

**Comment by liamdebeasi** [comment]
Thanks! This was due to a change in Stencil. For now, you can do `npm install @stencil/core@2.9.0`.

This issue was resolved via https://github.com/ionic-team/ionic-framework/pull/24152 and a fix will be available in the next Ionic 6 RC.

**Comment by liamdebeasi** [comment]
Hi everyone,

We just released Ionic 6.0.0-rc.2 with this fix.
